woocommerce/woocommerce-gutenberg-products-block
# Add support for the Editor Preview update when custom attributes are changed

⚠️ This PR is still WIP. It contains TypesScript errors and duplicated logic.


![Screen Capture on 2022-09-12 at 18-18-35](https://user-images.githubusercontent.com/4463174/189705501-b9a4d216-8f36-4541-8484-b4e643383128.gif)



This PR adds support for the editor preview update when custom attributes are changed. 

For example,  when `Show only products on sale` is enabled, the Query Loop block makes a GET request appending this query parameter:
```
__woocommerceVariationQuery%5BonSale%5D=true
```

In this way, with the filter `rest_product_query`, we can control the WordPress REST API, and we can create a custom WP_Query object by reading the query arguments. 


This PR fixes #6969 

#### Testing Steps

**Be sure that you are using [a Gutenberg build that contains this PR](https://github.com/WordPress/gutenberg/pull/44093).**

1. Add a Product Query block.
2. Click on “Show only products on sale”.
3. Notice that the block preview is updated and shows only products on sale.
